microservices/blast_query/blast_query/api.py

```diff
@@ -0,0 +1,70 @@
+import os
+
+from fastapi import FastAPI, HTTPException
+from blast_query.api_models import SequenceQuery
+from Bio.Blast import NCBIWWW
+import xmltodict
+
+app = FastAPI(
+    title="BLAST Query"
+)
+
+def choose_dataset(program):
+    if program in ["blastn", "tblastx", 'tblastn']:
+        return "nt"
+    elif program in ["blastp", "blastx"]:
+        return "nr"
+    else:
+        raise HTTPException(status_code=400, detail="Invalid program")
+
+def run_blast(program, query, descriptions=10, alignments=10, hitlist_size=10, expect=10.0):
+    NCBIWWW.email = os.getenv("EMAIL")
+
+    try:
+        dataset = choose_dataset(program)
+        result_handle = NCBIWWW.qblast(program, dataset, query, descriptions=descriptions, alignments=alignments, hitlist_size=hitlist_size, expect=expect)
+        result_dict = xmltodict.parse(result_handle.read())
+        result_handle.close()
+        return result_dict
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+@app.post("/blastn")
```

Comment:
Instead of many APIs I suggest to have 1 api with the type of blast that a user should use.

Reply:
Use a single endpoint /blast, pass type of blast to request model (as enum)